polaris-service/src/main/java/org/apache/polaris/service/task/TableCleanupTaskHandler.java

```diff
@@ -158,11 +161,98 @@ public boolean handleTask(TaskEntity cleanupTask) {
         for (PolarisBaseEntity createdTask : createdTasks) {
           taskExecutor.addTaskHandlerContext(createdTask.getId(), CallContext.getCurrentContext());
         }
+
+        // Schedule and dispatch prev metadata and stat files in seperated tasks
+        scheduleTableContentCleanupTask(
+            tableMetadata.previousFiles().stream().map(TableMetadata.MetadataLogEntry::file),
+            CleanupTableContentFileType.PREV_METADATA,
+            fileIO,
+            cleanupTask,
+            metaStoreManager,
+            polarisCallContext);
+        scheduleTableContentCleanupTask(
+            tableMetadata.statisticsFiles().stream().map(StatisticsFile::path),
+            CleanupTableContentFileType.STATISTICS,
+            fileIO,
+            cleanupTask,
+            metaStoreManager,
+            polarisCallContext);
```

Comment:
We could merge these two method calls. How about collecting the file first, then dispatching them into tasks for deletion? We don't have to distinguish file types while collecting them.